monitor/launch/launcher.go

```diff
@@ -29,37 +30,45 @@ const targetName = "cilium-node-monitor"
 
 // NodeMonitor is used to wrap the node executable binary.
 type NodeMonitor struct {
-	Arg     string
+	mutex   sync.RWMutex
+	arg     string
 	process *os.Process
 	state   *models.MonitorStatus
 }
 
 // Run starts the node monitor.
 func (nm *NodeMonitor) Run() {
 	for {
-		cmd := exec.Command(targetName, nm.Arg)
-		stdout, _ := cmd.StdoutPipe()
+		arg := nm.GetArg()
 
+		cmd := exec.Command(targetName, arg)
+		stdout, _ := cmd.StdoutPipe()
 		if err := cmd.Start(); err != nil {
 			log.Errorf("cmd.Start(): %s", err)
 		}
-		nm.process = cmd.Process
+
+		nm.setProcess(cmd.Process)
+		isProcNil := cmd.Process == nil
 
 		r := bufio.NewReader(stdout)
-		for nm.process != nil {
+		for !isProcNil {
```

Comment:
Add a function `processIsSet()` which calls `getProcess() != nil` so we don't need `isProcNil` and we don't need to update it inside the loop